### PR TITLE
Add boostchicken as codeowner for components

### PR DESCRIPTION
**Pull request by boostchicken**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/3870
**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by boostchicken** [comment]
@mx-psi more things i have recently put a ton of work on / have pull requests out for

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -72,6 +74,7 @@ processor/resourcedetectionprocessor/                @open-telemetry/collector-c
 processor/resourcedetectionprocessor/internal/azure  @open-telemetry/collector-contrib-approvers @mx-psi
 processor/routingprocessor/                          @open-telemetry/collector-contrib-approvers @jpkrohling
 processor/spanmetricsprocessor/                      @open-telemetry/collector-contrib-approvers @albertteoh
+processor/spanprocessor/                              @open-telemetry/collector-contrib-approvers @boostchicken
```
Comment (jpkrohling) [inline .github/CODEOWNERS:77]: I feel bad for nit-picking this, but there's an extra space here :-)

Comment (boostchicken) [reply .github/CODEOWNERS:77]: DOnt feel bad, details matter

Comment (boostchicken) [reply .github/CODEOWNERS]: Fixed!

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**